File: app/src/main/java/com/tristana/sandroid/ui/scheme/SchemeFilterActivity.java
package com.tristana.sandroid.ui.scheme;

import android.app.Activity;
import android.net.Uri;
import android.os.Bundle;

import com.therouter.router.Navigator;
import com.alibaba.android.arouter.facade.callback.NavCallback;
import com.therouter.TheRouter;

public class SchemeFilterActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Uri uri = getIntent().getData();
        TheRouter.build(uri).navigation(this, new NavCallback() {
            @Override
            public void onArrival(Navigator postcard) {
                finish();
            }
        });
    }
}

File: app/src/main/java/com/tristana/sandroid/ui/scheme/SchemeFilterActivity.kt
package com.tristana.sandroid.ui.scheme

import android.app.Activity
import android.os.Bundle
import com.blankj.utilcode.util.ActivityUtils
import com.therouter.TheRouter
import com.therouter.router.Route
import com.tristana.sandroid.MainActivity
import utils.router.RouterUtils

@Route(path = SchemeFilterActivity.ROUTE)
class SchemeFilterActivity : Activity() {

    companion object {
        const val ROUTE = "/app/activity/scheme"
    }

    var direct: String? = null
    var extra: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        TheRouter.inject(this)
        direct = intent.data?.getQueryParameter("direct")
        extra = intent.data?.getQueryParameter("extra")
        RouterUtils.routeWithDirect(MainActivity.ROUTE, direct, extra)
        ActivityUtils.finishActivity(this@SchemeFilterActivity)
    }
}